packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -436,6 +436,10 @@ class FlutterValidator extends DoctorValidator {
         frameworkVersion,
         _flutterRoot(),
       )));
+      messages.add(ValidationMessage(_userMessages.flutterUpstreamRepositoryUrl(version.repositoryUrl ?? 'unknown')));
```

Comment:
Actually, for users with the expected setup, this will add an extra line. I think we're only really interested in this value if it differs from whatever `flutterGit` evaluates to. Could you do something like:

```
final String configuredUpstream = version.repositoryUrl;
if (configuredUpstream != flutterGit) {
 
 messages.add(ValidationMessage(_userMessages.flutterUpstreamRepositoryUrl(configuredUpstream ?? 'unknown')));
}
```

This would require this PR: https://github.com/flutter/flutter/pull/79568/files, but you could merge that in to this PR.

Reply:
I think leaving this line will be fine, since it is displayed only by the `-v` flag. Also your comment [here](https://github.com/flutter/flutter/issues/78591#issuecomment-809569420) suggested `flutter doctor -v` to be a superset of `flutter --version`.

This kind of verbosity exists already:
```
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/anurag/.local/src/android_sdk <========
    • Platform android-30, build-tools 30.0.3
    • ANDROID_SDK_ROOT = /home/anurag/.local/src/android_sdk <========
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.
```

Reply:
Also, users would probably run `flutter doctor -v` than `flutter --version` while diagnosing.

Reply:
ahh yeah, good point, I forgot we already include the remote url in `flutter --version`. in that case we might as well include it.